cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -120,6 +122,105 @@ func NewFrontProxyClientCertAndKey(frontProxyCACert *x509.Certificate, frontProx
 	return frontProxyClientCert, frontProxyClientKey, nil
 }
 
+// UsingExternalCA determines whether the user is relying on an external CA.  We currently implicitly determine this is the case when the CA Cert
+// is present but the CA Key is not. This allows us to, e.g., skip generating certs or not start the csr signing controller.
+func UsingExternalCA(cfg *kubeadmapi.MasterConfiguration) bool {
+
+	if !ValidateCACert(cfg.CertificatesDir, kubeadmconstants.CACertAndKeyBaseName, "CA") {
+		return false
+	}
+
+	// Check CA Key is not present
+	caKeyPath := filepath.Join(cfg.CertificatesDir, kubeadmconstants.CAKeyName)
+	if _, err := os.Stat(caKeyPath); !os.IsNotExist(err) {
+		return false
+	}
+
+	if !ValidateSignedCert(cfg.CertificatesDir, kubeadmconstants.CACertAndKeyBaseName, kubeadmconstants.APIServerCertAndKeyBaseName, "API server") {
```

Comment:
Can you explain what you are suggesting a little more?  Do you mean this entire function?